tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <mock_nrf_rpc_transport.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#include <string.h>
+
+#define MAX_NUM_EXPECTED_PKTS 5
+
+typedef struct mock_nrf_rpc_tr_ctx {
+	const struct nrf_rpc_tr *transport;
+	nrf_rpc_tr_receive_handler_t receive_cb;
+	void *receive_ctx;
+
+	mock_nrf_rpc_pkt_t expected[MAX_NUM_EXPECTED_PKTS];
+	mock_nrf_rpc_pkt_t response[MAX_NUM_EXPECTED_PKTS];
+	size_t num_expected;
+	size_t cur_expected;
+
+	mock_nrf_rpc_pkt_t *cur_response;
+	struct k_work response_work;
+} mock_nrf_rpc_tr_ctx_t;
+
+static void response_send(struct k_work *work)
+{
+	mock_nrf_rpc_tr_ctx_t *ctx = CONTAINER_OF(work, mock_nrf_rpc_tr_ctx_t, response_work);
+	mock_nrf_rpc_pkt_t *response = ctx->cur_response;
+
+	ctx->receive_cb(ctx->transport, response->data, response->len, ctx->receive_ctx);
+}
+
+static int init(const struct nrf_rpc_tr *transport, nrf_rpc_tr_receive_handler_t receive_cb,
+		void *context)
+{
+	mock_nrf_rpc_tr_ctx_t *ctx = transport->ctx;
+
+	ctx->transport = transport;
+	ctx->receive_cb = receive_cb;
+	ctx->receive_ctx = context;
+
+	k_work_init(&ctx->response_work, response_send);
+
+	return 0;
+}
+
+static void log_payload(const char *caption, const uint8_t *payload, size_t length)
+{
+	/* Only log a prefix in case of large payloads. */
+	char payload_str[32];
+
+	if (!bin2hex(payload, length, payload_str, sizeof(payload_str))) {
+		length = (sizeof(payload_str) - sizeof("...")) / 2;
```

Comment:
Hm... I didn't get magic with subtraction and concatenation. Could you clarify a little bit?

Reply:
If the payload is very long, I want to display "..." at the end so I'm securing a space for that.

Reply:
Could you add some clarification comment to this?

Reply:
yes, sure